src/openagi/memory/base.py

```diff
@@ -101,3 +151,61 @@ def _create_metadata(self, task: Task) -> Dict[str, Any]:
             "task_description": task.description,
             "task_actions": task.actions,
         }
+
+    def add_ltm(self, session : SessionDict):
+        """
+        Add a session to the long term memory
+        :param session: The SessionDict object that has all the details of the session
+        :return: None
+        """
+        self.ltm_storage.save_document(
+            id = session.session_id,
+            document= session.query,
+            metadata= session.model_dump()
+        )
+        logging.info(f"Long term memory added for session : {session.session_id}")
+
+    def update_ltm(self, session: SessionDict) -> None:
+        """
+        Update an existing session in long-term memory.
+
+        :param session: The SessionDict object containing updated details of the session.
+        :return: None
+        """
+        self.ltm_storage.update_document(
+            id=session.session_id,
+            document=session.query,
+            metadata=session.model_dump()
+        )
+
+        logging.info(f"Long-term memory updated for session: {session.session_id}")
+
+    def get_ltm(self, query: str, n_results: int = 3) -> List[Dict[str, Any]]:
+        """
+        Retrieve and return the long-term memory based on a query.
+
+        :param query: The query string to search for.
+        :param n_results: The number of results to return.
+        :return: A dictionary of search results.
+        """
+        query_data = {
+            "query_texts": query,
+            "n_results": n_results,
+            "include": ["metadatas", "documents", "distances"],
+        }
+        response = self.ltm_storage.query_documents(**query_data)
+        results = []
+        # if "documents" in response and "distances" in response:
+        for doc, metadata, distance in zip(response["documents"][0], response["metadatas"][0], response["distances"][0]):
+            results.append({
+                "document": doc,
+                "metadata": metadata,
+                "similarity_score": 1 - distance
+            })
+        if results:
+            logging.info(f"Retrieved long-term memory for query: {query}\n{results[0]['document'][:250]}")
+            return results
+
+        logging.info(f"No documents found for query: {query}")
+        return results
+#
```

Comment:
should we need to have different file for LTM instead of having it in base.py? @gourab-aiplanet

Reply:
I think it makes sense for LTM to be in base, right now ltm is an attribute of memory object. If needed, we can try moving the ltm related functions to a different file, but it will complicate things since these functions are class functions of the memory object.